plugins/pulp_rpm/plugins/importers/yum/depsolve.py

```diff
@@ -421,8 +432,6 @@ def get_requirements(self, units):
         :rtype:     generator
         """
         for segment in paginate(units):
-            unit_ids = [unit.id for unit in segment]
-            fields = ['requires', 'id']
-            for result in models.RPM.objects.filter(id__in=unit_ids).only(*fields):
-                for require in result.requires or []:
+            for unit in segment:
+                for require in unit.requires or []:
```

Comment:
And, for completeness, here's the only place that `find_dependent_units` gets called, which is the only place that the method in question, `get_requirements`, is called:
https://github.com/seandst/pulp_rpm/blob/1432-revert-depsolve/plugins/pulp_rpm/plugins/importers/yum/associate.py#L170-L211

Note how the iterable being passed is a set of fully hydrated units that, were they going to blow up pulp, would have done so before `find_dependent_units` is called. :)